UnityProject/Assets/Scripts/Items/Implants/Organs/Liver.cs

```diff
@@ -37,44 +41,64 @@ public class Liver : BodyPartFunctionality
 		/// <summary>
 		/// Amount of reagents liver will attempt to pull from the blood. Affected by BodyPart efficiency
 		/// </summary>
-		[SerializeField] private float processAmount = 0.1f;
+		[SerializeField] private float processAmount = 2f;
 
 		/// <summary>
 		/// Multiplier to determine how many times as many reagents are flushed from the liver as were pulled from the blood stream.
-		/// This should be greater then one so the liver doesn't
+		/// This should be greater than one so the liver doesn't
 		/// </summary>
-		[SerializeField] private float flushMultiplier = 2;
+		[SerializeField] private float flushMultiplier = 1.1f;
 
+		[SerializeField] private float drunkMultiplier = 4;
+
+		private CirculatorySystemBase circ;
+		private StringBuilder debug;
+		private List<Tuple<Reagent, float>> tempArray;
+		private ReagentContainerBody blood;
+
+		public override void SetUpSystems()
+		{
+			circ = bodyPart.HealthMaster.GetComponent<CirculatorySystemBase>();
+			blood = RelatedPart.BloodContainer;
+			tempArray = new List<Tuple<Reagent, float>>();
+		}
 
 		public override void ImplantPeriodicUpdate()
 		{
 			//Liver has failed or just generally unable to process things, so don't let it.
 			if (RelatedPart.TotalModified == 0) return;
+			debug = new StringBuilder();
 
-			float tickPullProcessingAmnt =  RelatedPart.TotalModified *  processAmount;
-			float tickClearAmount = tickPullProcessingAmnt * flushMultiplier;
-
-			ReagentContainerBody blood = RelatedPart.BloodContainer;
+			BloodToLiver(blood);
+			Processing();
+			ReturnReagentsToBlood();
 
-			List<Tuple<Reagent,float>> tempArray = new List<Tuple<Reagent, float>>();
+			Logger.Log(debug.ToString(), Category.Health);
```

Comment:
Log and debug stringbuilder still needed? Comment them out if you think they'll be useful in the future

Reply:
it is currently logging an empty string, it is safe to just remove

Reply:
ah I opened my mouth before continuing reading